pallets/authors-manager/src/lib.rs

```diff
@@ -0,0 +1,634 @@
+        <AuthorActions<T>>::insert(
+            registered_author,
+            ingress_counter,
+            AuthorsActionData::new(
+                AuthorsActionStatus::AwaitingConfirmation,
+                tx_id,
+                AuthorsActionType::Activation,
+            ),
+        );
+    }
+
+    fn register_author(
+        author_account_id: &T::AccountId,
+        author_eth_public_key: &ecdsa::Public,
+    ) -> DispatchResult {
+        let decompressed_eth_public_key = decompress_eth_public_key(*author_eth_public_key)
+            .map_err(|_| Error::<T>::InvalidPublicKey)?;
+        let author_id_bytes =
+            <T as pallet::Config>::AccountToBytesConvert::into_bytes(author_account_id);
+        let function_name = b"addAuthor";
+
+        let params = vec![
+            (b"bytes".to_vec(), decompressed_eth_public_key.to_fixed_bytes().to_vec()),
+            (b"bytes32".to_vec(), author_id_bytes.to_vec()),
+        ];
+        let tx_id = <T as pallet::Config>::BridgeInterface::publish(
+            function_name,
+            &params,
+            PALLET_ID.to_vec(),
+        )
+        .map_err(|e| DispatchError::Other(e.into()))?;
+
+        let new_author_id = <T as SessionConfig>::ValidatorIdOf::convert(author_account_id.clone())
+            .ok_or(Error::<T>::ErrorConvertingAccountIdToAuthorId)?;
+
+        Self::start_activation_for_registered_author(author_account_id, tx_id);
+        T::ValidatorRegistrationNotifier::on_validator_registration(&new_author_id);
+        Self::deposit_event(Event::<T>::AuthorRegistered {
+            author_id: author_account_id.clone(),
+            eth_key: *author_eth_public_key,
+        });
+        Ok(())
+    }
+
+    fn get_ethereum_public_key_if_exists(account_id: &T::AccountId) -> Option<ecdsa::Public> {
+        return <EthereumPublicKeys<T>>::iter()
+            .filter(|(_, acc)| acc == account_id)
+            .map(|(pk, _)| pk)
+            .nth(0)
+    }
+
+    fn remove_ethereum_public_key_if_required(author_id: &T::AccountId) {
+        let public_key_to_remove = Self::get_ethereum_public_key_if_exists(&author_id);
+        if let Some(public_key_to_remove) = public_key_to_remove {
+            <EthereumPublicKeys<T>>::remove(public_key_to_remove);
+        }
+    }
+
+    /// We assume the full public key doesn't have the `04` prefix
+    #[allow(dead_code)]
+    fn compress_eth_public_key(full_public_key: H512) -> ecdsa::Public {
+        let mut compressed_public_key = [0u8; 33];
+
+        // Take bytes 0..32 from the full plublic key ()
+        compressed_public_key[1..=32].copy_from_slice(&full_public_key.0[0..32]);
+        // If the last byte of the full public key is even, prefix compresssed public key with 2,
+        // otherwise prefix with 3
+        compressed_public_key[0] = if full_public_key.0[63] % 2 == 0 { 2u8 } else { 3u8 };
+
+        return ecdsa::Public::from_raw(compressed_public_key)
+    }
```

Comment:
This seems like something that should live in the primitives, or remove if it is dead code.